File: translations/codenet/deepseek_commented/C2P/codellama/Code/s310320755.py
#include<stdio.h>

# Infinite loop to keep the program running until n is 0
while True:
    n = int(input("Enter the size of the magic square: "))
    if n == 0:
        break

    # Initialize the starting position and the counter
    flag = 1
    r = (n + 1) // 2 + 1
    c = (n + 1) // 2
    arr = [[0 for i in range(n + 1)] for j in range(n + 1)]
    arr[r][c] = cnt = 1

    # Loop to fill the magic square
    while cnt != n * n + 1:
        cnt += 1
        if flag:
            r += 1
            c += 1
            if c > n:
                c = 1
            if r > n:
                r = 1
            if arr[r][c]:
                flag = 0
                cnt -= 1
            else:
                arr[r][c] = cnt
        else:
            r += 1
            c -= 1
            if c < 1:
                c = n
            if r > n:
                r = 1
            if arr[r][c]:
                flag = 0
            else:
                arr[r][c] = cnt
                flag = 1

    # Print the generated magic square
    for r in range(1, n + 1):
        for c in range(1, n + 1):
            print(arr[r][c], end=" ")
        print()

# 

File: translations/codenet/mistral_commenetd/C2J/deepseek/Code/s310320755.java
import java.util.Scanner;

public class MagicSquare {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n;
        while (true) {
            System.out.print("Enter the size of the square matrix: ");
            n = scanner.nextInt();
            if (n == 0) {
                break;
            }
            int[][] arr = new int[16][16];
            int r = (n + 1) / 2 + 1;
            int c = (n + 1) / 2;
            int cnt = 1;
            boolean flag = true;
            arr[r][c] = cnt;
            while (cnt != n * n + 1) {
                cnt++;
                if (flag) {
                    r++;
                    c++;
                    if (c > n) {
                        c = 1;
                    }
                    if (r > n) {
                        r = 1;
                    }
                    if (arr[r][c] != 0) {
                        flag = false;
                        cnt--;
                    } else {
                        arr[r][c] = cnt;
                    }
                } else {
                    r++;
                    c--;
                    if (c < 1) {
                        c = n;
                    }
                    if (r > n) {
                        r = 1;
                    }
                    if (arr[r][c] != 0) {
                        flag = true;
                    } else {
                        arr[r][c] = cnt;
                        flag = true;
                    }
                }
            }
            for (r = 1; r < n + 1; r++) {
                for (c = 1; c < n + 1; c++) {
                    System.out.printf("%4d", arr[r][c]);
                    arr[r][c] = 0;
                    if (c == n) {
                        System.out.println();
                    }
                }
            }
        }
    }
}
